crates/isograph_schema/src/isograph_schema.rs

```diff
@@ -42,6 +42,17 @@ pub trait SchemaValidationState: Debug {
     ///   i.e. DefinedField<ServerFieldId, ClientFieldId>
     type ClientFieldSelectionScalarFieldAssociatedData: Debug;
 
+    /// The associated data type of scalars in client pointers' selection sets and unwraps
```

Comment:
Let's take a second to reconsider... I think I gave you bad instructions.

1. These comments are somewhat outdated: `DefinedField` no longer exists, it's now `ValidatedScalarFieldAssociatedData` (which contains `FieldType`, which is what `DefinedField` has become). Let's change `ObjectId` (since renamed to `ServerObjectId`) to `ValidatedLinkedFieldAssociatedData`
2. I don't think we need an additional type param here. What do you think?

What we're doing when we're validating is that we're going from something that is selected as a scalar (i.e. has no curlies) or something selected as a linked field (has curlies), and going from no information other than the name of the field to an enum of all the possible things it could be. (Plus some other data that we can only get while validating... but let's ignore that for now.)

So, a scalar selection goes from no info to either a client field or a server scalar field.

A linked selection goes from no info to (currently) just an `ServerObjectId`. But now it should go to an enum with either `ServerObjectId` (i.e. server linked field) or client pointer.

Soooo.... I don't think `FieldType` needs a third variant. Instead, as a first step, we should replace `ServerObjectId` in `ValidatedLinkedFieldAssociatedData` to a `FieldType<ServerObjectId, ClientPointerId>`

----

I think that's the correct move, given our current setup.